vllm/v1/worker/gpu_model_runner.py

```diff
@@ -547,6 +554,55 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
         if batch_changed or batch_reordered:
             self.input_batch.refresh_sampling_metadata()
 
+    def _add_multimodal_inputs_to_model_args(
+            self,
+            model_kwargs: dict[str, Any],
+            scheduler_output: "SchedulerOutput",
+            num_reqs: int = -1):
+        # Multi-modal data.
+        if scheduler_output:
+            multi_modal_kwargs_list = []
+            for req in scheduler_output.scheduled_new_reqs:
+                req_mm_inputs = req.mm_inputs
+                if not isinstance(req_mm_inputs, list):
+                    req_mm_inputs = list(req_mm_inputs)
+                multi_modal_kwargs_list.extend(req_mm_inputs)
+            multi_modal_kwargs = MultiModalKwargs.batch(
+                multi_modal_kwargs_list)
+        else:
+            # The only case where SchedulerOtput is None is for a dummy run,
+            # let's get some dummy data.
+            dummy_data = [
+                self.mm_registry.get_decoder_dummy_data(
+                    model_config=self.model_config, seq_len=1).multi_modal_data
+                for i in range(num_reqs)
+            ]
+            multi_modal_kwargs = MultiModalKwargs.batch(dummy_data)
+
+        model_kwargs.update(multi_modal_kwargs)
+
+    def _maybe_add_multimodal_kwargs(
+            self,
+            model_kwargs: dict[str, Any],
+            scheduler_output: "SchedulerOutput" = None,
+            num_reqs: int = -1):
+
+        if self.model_supports_multimodal_raw_input:
+            self._add_multimodal_inputs_to_model_args(model_kwargs,
+                                                      scheduler_output,
+                                                      num_reqs)
+
+    def _maybe_compute_attn_prefix(
```

Comment:
What is the purpose of this function? It looks like an extension point for a subclass but it's not called anywhere.